### PR TITLE
Add manpages for pmemkv_config functions

### PR DESCRIPTION
**Pull request by igchor**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemkv/340)

### CODE REVIEW

**Comment by szyrom** [comment]
@George330 please review

File: doc/libpmemkv.3.md [hunk 1]
```diff
@@ -48,14 +48,162 @@ date: pmemkv version 0.8
...
+
+Puts value `value` to pmemkv_config at key `key`. 
+
+`int pmemkv_config_put_string(pmemkv_config *config, const char *key, const char *value);`
+
+Puts null-terminated string to pmemkv_config.
+
+`int pmemkv_config_put_data(pmemkv_config *config, const char *key, const void *value, size_t value_size);`
+
+Puts copy of binary data pointed to by `value` to pmemkv_config. `value_size`
```
Comment (marcinslusarz) [inline doc/libpmemkv.3.md]: s/to//?

File: doc/libpmemkv.3.md [hunk 2]
```diff
@@ -48,14 +48,162 @@ date: pmemkv version 0.8
...
+specifies size of the data.
+
+`int pmemkv_config_put_object(pmemkv_config *config, const char *key, void *value, void (*deleter)(void *));`
+
+Puts pointer `value` to pmemkv_config. `value` can point to any arbitrary
```
Comment (marcinslusarz) [inline doc/libpmemkv.3.md]: any or arbitrary - not both :)

File: doc/libpmemkv.3.md [hunk 3]
```diff
@@ -48,14 +48,162 @@ date: pmemkv version 0.8
...
+object. `deleter` parameter specified function which is called for `value`
```
Comment (marcinslusarz) [inline doc/libpmemkv.3.md]: specifies function which will be called

File: doc/libpmemkv.3.md [hunk 4]
```diff
@@ -48,14 +48,162 @@ date: pmemkv version 0.8
...
+when config is destroyed (using pmemkv_config_delete).
+
+`int pmemkv_config_get_uint64(pmemkv_config *config, const char *key, uint64_t *value);`
+`int pmemkv_config_get_int64(pmemkv_config *config, const char *key, int64_t *value);`
+`int pmemkv_config_get_double(pmemkv_config *config, const char *key, double *value);`
+
+Gets value of element with key `key`. Value is copied to variable pointed to by
```
Comment (marcinslusarz) [inline doc/libpmemkv.3.md]: "to by"?

File: doc/libpmemkv.3.md [hunk 5]
```diff
@@ -48,14 +48,162 @@ date: pmemkv version 0.8
...
+Gets pointer to binary data. Data is not copied. After successful call
+`value` points to data stored in pmemkv_config and `value_size` holds size of the data.
+
+`int pmemkv_config_get_object(pmemkv_config *config, const char *key, const void **value);`
+
+Gets pointer to an object. After successful call, `value` points the object.
+
+`int pmemkv_config_from_json(pmemkv_config *config, const char *jsonconfig);`
+
+Parses JSON string and puts all elements found in JSON in `config`. Allowed types
```
Comment (marcinslusarz) [inline doc/libpmemkv.3.md]: in->into?

File: doc/libpmemkv.3.md [hunk 6]
```diff
@@ -48,14 +48,162 @@ date: pmemkv version 0.8
...
+in JSON strings and their corresponding types in pmemkv_config are:
+* `number` -> int64 for integral types and double for floating point,
+* `string` -> const char *,
+* `object` (another JSON string) -> pointer to pmemkv_config (can be obtained using pmemkv_config_get_object)
+* `True`, `False` -> int64
+
+Element stored in pmemkv_config, which were putted using a specific function can be obtained
```
Comment (marcinslusarz) [inline doc/libpmemkv.3.md]: put/put/put

File: doc/libpmemkv.3.md [hunk 7]
```diff
@@ -48,14 +48,162 @@ date: pmemkv version 0.8
...
+only using corresponding pmemkv_config_get_ function (for example, elements putted using pmemkv_config_put_double
+can only get obtained using pmemkv_config_get_double). Exception from this rule
+are functions for uint64 and int64. If value putted by pmemkv_config_put_int64 is in uint64_t range
```
Comment (marcinslusarz) [inline doc/libpmemkv.3.md]: .

File: doc/libpmemkv_config.3.md [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+[SEE ALSO](#see-also)<br />
+
+
+# NAME #
+
+**pmemkv** - Key/Value Datastore for Persistent Memory
+
+# SYNOPSIS #
+
+```
```
Comment (lukaszstolarczuk) [inline doc/libpmemkv_config.3.md]:
```c

**Pull request merged**